src/libraries/System.Numerics.Vectors/tests/GenericVectorUnsupportedTests.cs

```diff
@@ -358,5 +359,29 @@ public void AsToTest()
             Vector<int> vector = default;
             Assert.Throws<NotSupportedException>(() => vector.As<int, bool>());
         }
+
+        [Fact]
+        public void IsNotSupportedBoolean() => TestIsNotSupported<bool>();
+
+        [Fact]
+        public void IsNotSupportedChar() => TestIsNotSupported<char>();
+
+        [Fact]
+        public void IsNotSupportedHalf() => TestIsNotSupported<Half>();
+
+        [Fact]
+        public void IsNotSupportedInt128() => TestIsNotSupported<Int128>();
+
+        [Fact]
+        public void IsNotSupportedUInt128() => TestIsNotSupported<UInt128>();
+
+        private static void TestIsNotSupported<T>()
+            where T : struct
+        {
+            Assert.False(Vector<T>.IsSupported);
+
+            MethodInfo methodInfo = typeof(Vector<T>).GetProperty("IsSupported", BindingFlags.Public | BindingFlags.Static).GetMethod;
+            Assert.False((bool)methodInfo.Invoke(null, null));
```

Comment:
Do all these tests rely on the specifics of the machine they are running on? I'm assuming they are set up to pass regardless of the machine, so can you elaborate on how that is being accomplished?

Reply:
The `IsHardwareAccelerated` depends on the machine they run on, the `IsSupported` ones do not and report static `true` or `false` depending on the `T` having some support, even software based.

Reply:
This test in particular is just validating that the indirect invocation (e.g. calling through reflection) works as expected and returns the same value as direct invocation. This is important since some of these APIs are "intrinsic" and may accidentally behave differently between the two otherwise.